api/models.py

```diff
@@ -228,3 +229,54 @@ class Tag(ModelReprMixin, models.Model):
 
     def __str__(self):
         return self.title
+
+
+class Infraction(ModelReprMixin, models.Model):
+    """An infraction for a member of our Discord server."""
+
+    TYPE_CHOICES = (
+        ("note", "Note"),
+        ("warning", "Warning"),
+        ("mute", "Mute"),
+        ("ban", "Ban"),
+        ("kick", "Kick"),
+        ("superstar", "Superstar")
+    )
+    id = models.UUIDField(
+        primary_key=True,
+        default=uuid.uuid4,
+        help_text="The UUID of the infraction."
+    )
+    inserted_at = models.DateTimeField(
+        auto_now_add=True,
+        help_text="The date and time of the creation of this infraction."
+    )
+    expires_at = models.DateTimeField(
+        null=True,
+        help_text=(
+            "The date and time of the expiration of this infraction."
+            "Null if the infraction is permanent or it can't expire."
+        )
```

Comment:
A validator for this being in the future might make sense.

Reply:
In the sense that it shouldn't allow an expiration for certain types that can't expire (e.g. kick or warn)?

Reply:
Oh I misread, but I think my idea is good.
Also, why would we validate if an expiration is in the future. Isn't that the point of an expiration - to be at a later date?

Reply:
> In the sense that it shouldn't allow an expiration for certain types that can't expire (e.g. kick or warn)?
 
Yeah, the expiry should either be `None` or a datetime in the future, IMO.

> Also, why would we validate if an expiration is in the future. Isn't that the point of an expiration - to be at a later date?
 
Yes, that's kind of what I want validations to cover: impossible states. After all, the Django API is the gateway to our database, and we aren't placing the constraints directly on the database (we probably should, but that's out of scope for now). I also added validators that check whether snowflakes are positive and permissions are between 0 and the max permission bitset, for instance.

However, using a validator like that will pretend us from importing infractions from the old site to the new site, so we probably shouldn't do that, or only put it in later.

Reply:
So for now, I think this is fine.